### PR TITLE
Add the NoDrift dummy drift detector

### PR DESCRIPTION
**Pull request by smastelini**
This PR addresses #1332 by adding the `drift.NoDrift` class. This detector does nothing and can be used to bypass the concept drift handling capabilities in ARF.

### CODE REVIEW

**Review by MaxHalford — APPROVED** [review]

File: river/drift/no_drift.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+from river import base
+from river.base.drift_detector import DriftDetector
+
+
+class NoDrift(base.DriftDetector):
+    """Dummy class used to turn off concept drift detection capabilities of adaptive models.
+
+
```
Comment (MaxHalford) [inline river/drift/no_drift.py:10]:
```suggestion

```

File: river/drift/no_drift.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    It always signals that no concept drift was detected.
+
+
```
Comment (MaxHalford) [inline river/drift/no_drift.py:13]:
```suggestion

```

File: river/drift/no_drift.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    Examples
+    --------
+    >>> from river import drift
```
Comment (MaxHalford) [inline river/drift/no_drift.py:16]:
```suggestion
    --------
    
    >>> from river import drift
```

File: river/forest/adaptive_random_forest.py [hunk 4]
```diff
@@ -119,20 +118,18 @@ def n_warnings_detected(self, tree_id: int | None = None) -> int | None:
...
+        if self._warning_detection_disabled:
+            return 0
 
         if tree_id is None:
             return sum(self._warning_tracker.values())
 
         return self._warning_tracker[tree_id]
 
-    def n_drifts_detected(self, tree_id: int | None = None) -> int | None:
+    def n_drifts_detected(self, tree_id: int | None = None) -> int:
```
Comment (MaxHalford) [inline river/forest/adaptive_random_forest.py:129]: Nice

Comment (smastelini) [reply river/forest/adaptive_random_forest.py:129]: :D

**Pull request merged**